### PR TITLE
Minor Bug Fixes

### LINKED ISSUE #6: Retrieving protected objects with no present filesets causes error
When processing the protected objects, the module mostly does null checks around everything.  This check isn't applied to filesets - therefore, it causes the report to error out due to the following error

```
Cannot validate argument on parameter 'id'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
```

Need to apply the same logic to filesets as all other protected objects.

### LINKED ISSUE #8: Output Version of all modules in the Verbose logging
Need to output version of AsBuiltReport.Core, AsBuiltReport.Rubrik.CDM and Rubrik Powershell modules within the verbose logging.  Will make troubleshooting customer issues a lot easier to know what versions they are running without having to ask :)

### PR DESCRIPTION
**Pull request by mwpreston**
A couple of minor bug fixes around count checks and vCD errors. 

Fixed issue around code error when no filesets or NAS Shares are present

Resolve #6 Resolve #8

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Rubrik CDM As Built Report Changelog
 
+### Modified
```
Comment (tpcarman) [inline CHANGELOG.md:20]: Need to increase the version number to allow updates to be published to PS Gallery. You will also need to modify the version in the manifest.

File: Src/Public/Invoke-AsBuiltReport.Rubrik.CDM.ps1 [hunk 2]
```diff
@@ -433,7 +433,7 @@ function Invoke-AsBuiltReport.Rubrik.CDM {
                                 @{N = "Username"; E = { $_.username } },
                                 @{Name = "VM Linking"; Expression = { $_.conflictResolutionAuthz } },
                                 @{Name = "Certificate"; E = { $_.caCerts } }
-                                Write-Verbose -Message "[Rubrik] [$($brik)] [Cluster Settings] Retrieving vCloud Directory Information"
+                                Write-Verbose -Message "[Rubrik] [$($brik)] [Cluster Settings] Retrieving vCloud Director Information"
```
Comment (tpcarman) [inline Src/Public/Invoke-AsBuiltReport.Rubrik.CDM.ps1]: I'd like to switch `Write-Verbose` to [`Write-PScriboMessage`](https://github.com/iainbrighton/PScribo/blob/dev/Src/Public/Write-PScriboMessage.ps1) to keep the verbose output format consistent.

File: Src/Public/Invoke-AsBuiltReport.Rubrik.CDM.ps1 [hunk 3]
```diff
@@ -1089,12 +1089,18 @@ function Invoke-AsBuiltReport.Rubrik.CDM {
...
-                            @{N = "Operating System"; E = { $_.operatingSystemType } }, @{N = "Fileset Name"; E = { $_.name } },
-                            @{N = "SLA Domain"; E = { $_.effectiveSlaDomainName } }
+                            $Filesets = Get-RubrikFileset -Relic:$false -PrimaryClusterID local | Where-Object { $_.effectiveSlaDomainId -ne 'Unprotected' -and $null -eq $_.shareId }
+                            if ($null -ne $Filesets -and 0 -ne $Filesets[0].total ) {
+                                $Filesets = Get-RubrikFileset -Relic:$false -PrimaryClusterID local -DetailedObject | Where-Object { $_.effectiveSlaDomainId -ne 'Unprotected' -and $null -eq $_.shareId } | Select-Object -Property @{N = "Hostname"; E = { $_.hostname } },
+                                @{N = "Operating System"; E = { $_.operatingSystemType } }, @{N = "Fileset Name"; E = { $_.name } },
+                                @{N = "SLA Domain"; E = { $_.effectiveSlaDomainName } }
```
Comment (tpcarman) [inline Src/Public/Invoke-AsBuiltReport.Rubrik.CDM.ps1:1106]: Look to move away from using expressions in favour of `PSCustomPObjects` in alignment with the contributing [guidelines](https://github.com/AsBuiltReport/AsBuiltReport.Core/blob/dev/CONTRIBUTING.md#use-a-consistent-coding-style)

**Review by tpcarman — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (mwpreston) [reply CHANGELOG.md:20]: Done!

File: Src/Public/Invoke-AsBuiltReport.Rubrik.CDM.ps1 (hunk 2 above)
Comment (mwpreston) [reply Src/Public/Invoke-AsBuiltReport.Rubrik.CDM.ps1]: And done!

File: Src/Public/Invoke-AsBuiltReport.Rubrik.CDM.ps1 (hunk 3 above)
Comment (mwpreston) [reply Src/Public/Invoke-AsBuiltReport.Rubrik.CDM.ps1:1106]: Will tackle this in another PR as it's probably a big change!  I'll create an issue for it!

**Review by tpcarman — APPROVED** [review]

**Pull request merged**